test/framework/bootstrap/kind_util.go

```diff
@@ -62,10 +62,14 @@ func CreateKindBootstrapClusterAndLoadImages(ctx context.Context, input CreateKi
 	clusterProvider.Create(ctx)
 	Expect(clusterProvider.GetKubeconfigPath()).To(BeAnExistingFile(), "The kubeconfig file for the kind cluster with name %q does not exists at %q as expected", input.Name, clusterProvider.GetKubeconfigPath())
 
-	LoadImagesToKindCluster(ctx, LoadImagesToKindClusterInput{
+	err := LoadImagesToKindCluster(ctx, LoadImagesToKindClusterInput{
 		Name:   input.Name,
 		Images: input.Images,
 	})
+	if err != nil {
+		clusterProvider.Dispose(ctx)
```

Comment:
err := ?
Otherwise I do expect the following statement to always fail

Reply:
I don't understand what you mean

Reply:
I find this bit of code a little bit confusing, because we are entering in a branch where `err != nil`, and then run `Expect(err).NotTo(HaveOccurred())` than will always fail. 
So my first reaction was that it was missing err := in front of `clusterProvider.Dispose()`

Now, re-reading the code I understand your intention is to handle the error, but not to shallow it.
That's fine, but could you kindly add a comment like "if there is an error, cleanup the kind cluster before returning the error"

Reply:
Sure, yeah I figured that was clear